### DISCUSSION TITLE
What is the best way to use Serilog and ELK stack in .NET Aspire?

### DISCUSSION THREAD

**Discussion by MarcinKruszynski** [microsoft/aspire#5211] [Q&A]
Hi,

we are starting with .NET Aspire project and we want to use ELK stack for structure logs. Everything will be deployed to Kubernetes cluster.  It seems that example for Serilog and ELK stack in .NET Aspire could be useful. 

I know that some ELK things are at proposal stage https://github.com/dotnet/aspire/discussions/1570, f.e Kibana or Logstash support. 

I also know about Elasticsearch support in .NET Aspire 8.1. It is only for Elasticsearch hosting and client, without Serilog that has dedicated nuget package like Elastic.Serilog.Sinks (which is for every .NET app, not only .NET Aspire). 

1. Is it better use optional Logstash and send logs to http endpoint like https://learn.microsoft.com/en-us/dotnet/architecture/cloud-native/logging-with-elastic-stack  or try to use Elastic.Serilog.Sinks extension directly from .NET Aspire microservices?  Elastic.Serilog.Sinks seems not to use latest Elasticsearch .NET Aspire client component.  Or is it necessary to write custom component for .NET Aspire similar to Elastic.Serilog.Sinks? 

2. It seems that Kibana extension could be useful for Elasticsearch hosting like f.e Kafka UI extension for Kafka. Do you have any example how to write custom extension with Kibana docker image.

**Comment by davidfowl** [comment]
> It seems that Kibana extension could be useful for Elasticsearch hosting like f.e Kafka UI extension for Kafka. Do you have any example how to write custom extension with Kibana docker image.

https://learn.microsoft.com/en-us/dotnet/aspire/extensibility/custom-hosting-integration
